### PR TITLE
Add whitelisting CIDR Blocks

### PR DESCRIPTION
**Pull request by gauth-fr**
I added a small feature to whitelist CIDRs block, even if they are supposed to be rejected (non-allowed country)
The provided example is the google cidr.

### CODE REVIEW

**Comment by gauth-fr** [comment]
and that should resolve https://github.com/nscuro/traefik-plugin-geoblock/issues/22 ?

File: plugin.go [hunk 1]
```diff
@@ -179,3 +194,34 @@ func (p Plugin) Lookup(ip string) (string, error) {
...
 }
+
+func initAllowedIPBlocks(allowedIPBlocks []string) []*net.IPNet {
+
+	var allowedIPBlocksNet []*net.IPNet
+
+	for _, cidr := range allowedIPBlocks {
+		_, block, err := net.ParseCIDR(cidr)
+		if err != nil {
+			panic(fmt.Errorf("parse error on %q: %v", cidr, err))
```
Comment (nscuro) [inline plugin.go:205]: Instead of `panic`king, this function should return an `error`.

File: plugin.go [hunk 2]
```diff
@@ -154,10 +159,20 @@ func (p Plugin) CheckAllowed(ip string) (bool, string, error) {
 	var allowed bool
 	for _, allowedCountry := range p.allowedCountries {
 		if allowedCountry == country {
-			allowed = true
-			break
+			// allowed = true
+			return true, country, nil
+			//break
```
Comment (nscuro) [inline plugin.go:164]: I think the comments can be removed?

```suggestion
			return true, country, nil
```

File: plugin.go [hunk 3]
```diff
@@ -154,10 +159,20 @@ func (p Plugin) CheckAllowed(ip string) (bool, string, error) {
...
 		}
 	}
+
+	allowed, err = p.isAllowedIPBlocks(ip)
+	if err != nil {
+		return false, "", fmt.Errorf("checking if %s is part of an allowed range failed: %w", ip, err)
+	}
+	if allowed {
+		return true, country, nil
+	}
```
Comment (nscuro) [inline plugin.go:174]: This is redundant and can be removed.

```suggestion
```

**Review by nscuro — CHANGES_REQUESTED** [review]
Thanks for the PR @gauth-fr! 🚀

Some minor nitpicks, then we're good to go.

**Comment by gauth-fr** [comment]
Sure I agree :) what should I do? (I'm quite new to git and go^^)
Change my code and commit again?

**Comment by nscuro** [comment]
Oh sorry, based on your code I was assuming you're experienced! :)

Yeah, it boils down to changing your code locally according to my comments, committing and pushing your changes. The PR will be updated automatically, and we can mark my comments as "resolved". If you run into issues, I'm happy to assist. Just tell me what tools you use to work with Git, what platform (Windows, Linux, macOS) you're on, and I can guide you.

But you may notice that two of my comments have *suggested changes* in them. You can apply my suggestions using the "commit suggestion" button:

Screenshot 2022-11-04 at 21 30 42

It will create a commit for you.

**Comment by gauth-fr** [comment]
Nah I learned go like 4h ago thanks to Google and stackoverflow^^
I'll give a try tomorrow, it starts to be late to geek ^^

**Comment by gauth-fr** [comment]
Done!
Then i was wondering if, in case of CIDR parsing error i should really return an error or just print an error message.
At least it wouldn't block everything no?

**Comment by nscuro** [comment]
> in case of CIDR parsing error i should really return an error or just print an error message.

I think that when there is a configuration error, it should be *super* obvious that something is not right. I doubt many users follow their logs, and even if they do, a warning may be easy to miss.

**Review by nscuro — APPROVED** [review]
Thanks again for the PR!

**Comment by gauth-fr** [comment]
Yeah! My first approved PR^^

**Pull request merged**